File: supervision-client/src/app/shared/technical/reporting/report-command.ts
import { ReportEntity } from "./report-entity";
import { SearchCriteria } from "./search-criteria";
import { SortDirection } from "./sort-direction";

export class ReportCommand {
    criterias: SearchCriteria[] = [];
    sortFieldName: string = "id";
    sortDirection: SortDirection = SortDirection.ASC;
    page: number = 0;
    nbItemsPerPage: number = 100;
}

File: supervision-server-data/src/main/java/supervision/server/reporting/command/ReportCommand.java
package supervision.server.reporting.command;

import java.util.List;

import org.springframework.data.domain.Sort.Direction;

import lombok.Data;

@Data
public class ReportCommand {

	public static final String ID_FIELD_NAME = "id";

	public static final int NB_MAX_RESULTS = 200;

	private ReportEntity reportEntity;
	
	private List<SearchCriteria> criterias;
	
	private String sortFieldName;
	
	private Direction sortDirection = Direction.ASC;

	private int page = 0;
	
	private int nbItemsPerPage = NB_MAX_RESULTS;

}
